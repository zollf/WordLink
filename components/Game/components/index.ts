export { default as Word } from './Word';
export { default as Letter } from './Letter';
export { default as LetterSelection } from './LetterSelection';
export { default as CurrentWord } from './CurrentWord';
export { default as Path } from './Path';
export { default as ControlButtons } from './ControlButtons';
export { default as Completed } from './Completed';