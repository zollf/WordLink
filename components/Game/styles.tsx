import { StyleSheet } from "react-native";
import { heightPercentageToDP } from 'react-native-responsive-screen';

const styles = StyleSheet.create({
  game: {
    width: "100%",
    height: "100%",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
  },
  information: {
    position: "absolute",
    top: heightPercentageToDP('8%'),
  },
  content: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    marginBottom: 80,
  },
  letterSelection: {
    height: 150,
    position: "absolute",
    bottom: heightPercentageToDP('15%'),
  },
  completed: {
    position: "absolute",
    bottom: heightPercentageToDP('15%'),
  },
});

export default styles;