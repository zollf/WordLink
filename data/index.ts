export { default as easy } from './easy';
export { default as medium } from './medium';
export { default as hard } from './hard';
export { default as dictionary } from './dictionary';
export { default as getLevel } from './getLevel';
