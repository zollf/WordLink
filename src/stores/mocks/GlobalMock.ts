const GlobalMock: Global = {
  _ready: true,
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
  setUserInfo: jest.fn(),
  userInfo: {
    username: 'jeff',
    points: 20,
    puzzles: {
      completed: 2,
      easy: [],
      medium: [],
      hard: [],
    },
    hintsUsed: 6,
    hardestDifficulty: 'Medium',
    starsEarnt: 69,
  },
  init: jest.fn(),
};

export default GlobalMock;
