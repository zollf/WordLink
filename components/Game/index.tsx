import React, { useState, useEffect } from 'react';
import { View, Animated } from 'react-native';
import { observer } from 'mobx-react';
import { useStore, useAnimate } from '../../hooks';
import CircleButton from '../CircleButton';

import { 
  Word, 
  LetterSelection, 
  CurrentWord, 
  ControlButtons, 
  Path,
  Completed,
  ExitModal,
  Information
} from './components';

import { ElasticSlideIn, FadeSlideIn, Animate } from '../../animations';
import styles from './styles';

const Game = () => {
  const { game, clearGame, completed } = useStore('gameStore');
  const { setCurrentPage } = useStore('global');
  const [exitModal, setExitModal] = useState(false);

  const [infoT, infoO] = useAnimate(-100, 0);
  const [word1T, word1O] = useAnimate(-100, 0);
  const [pathT, pathO] = useAnimate(-100, 0);
  const [currentT, currentO] = useAnimate(-100, 0);
  const [word2T, word2O] = useAnimate(-100, 0);
  const [lettersT, lettersO] = useAnimate(-100, 0);
  useEffect(() => {
    Animated.parallel([
      Animated.stagger(20, [
        ElasticSlideIn(infoT),
        ElasticSlideIn(word1T),
        ElasticSlideIn(pathT),
        ElasticSlideIn(currentT),
        ElasticSlideIn(word2T),
        ElasticSlideIn(lettersT),
      ]),
      Animated.stagger(20, [
        FadeSlideIn(infoO),
        FadeSlideIn(word1O),
        FadeSlideIn(pathO),
        FadeSlideIn(currentO),
        FadeSlideIn(word2O),
        FadeSlideIn(lettersO),
      ]),
    ]).start();
  });

  const handleExit = () => {
    setCurrentPage('menu');
    clearGame();
  };

  return (
    <View style={styles.game}>
      {exitModal && <ExitModal close={() => setExitModal(false)} exit={handleExit}/>}
      <View style={styles.back}>
        <CircleButton overrideCallback={() => setExitModal(true)}/>
      </View>
      
      
      <Animate style={styles.information} transform={infoT} opacity={infoO}>
        <Information/>
      </Animate>

      <View style={styles.content}>
        <Animate transform={word1T} opacity={word1O}>
          <Word color="orange" word={game.start} />
        </Animate>
        
        <Animate transform={pathT} opacity={pathO}>
          <Path />
        </Animate>
        
        <Animate transform={currentT} opacity={currentO}>
          <CurrentWord />
        </Animate>

        <Animate transform={word2T} opacity={word2O}>
          <Word color="blue" word={game.end} />
        </Animate>
      </View>

      <Animate style={styles.letterSelection} transform={lettersT} opacity={lettersO}>
        <LetterSelection />
      </Animate>

      {!completed ? (
        <ControlButtons /> 
      ) : ( 
        <View style={styles.completed}>
          <Completed />
        </View>
      )}
    </View>
  );
};

export default observer(Game);