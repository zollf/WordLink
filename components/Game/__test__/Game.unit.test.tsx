import React from 'react';
import { Provider } from 'mobx-react';
import GlobalMock from '../../../stores/mocks/GlobalMock';
import GameMock from '../../../stores/mocks/GameMock';
import { getLevel } from '../../../data';

import Game from '../';

import renderer from 'react-test-renderer';

GameMock.id = 1;
GameMock.difficulty = 'easy';
GameMock.inGame = true;

const game = getLevel('easy')![1].game;
GameMock.game = game;
GameMock.currentWord = game.start;
GameMock.path = [game.start];

const component = (
  <Provider global={GlobalMock} gameStore={GameMock}>
    <Game />
  </Provider>
);

describe('<Game />', () => {
  const wrapper = renderer.create(component);
  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});