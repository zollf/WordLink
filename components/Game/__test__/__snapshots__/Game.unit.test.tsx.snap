// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game /> matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "height": "100%",
        "left": 37.5,
        "position": "absolute",
        "top": 120,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4284860671,
              4282432255,
            ],
            "endPoint": undefined,
            "locations": Array [
              0.5,
              0.5,
            ],
            "startPoint": Array [
              -0.1,
              0.1,
            ],
          }
        }
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 17,
            "display": "flex",
            "height": 60,
            "justifyContent": "center",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
            "width": 60,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
          "top": 106.5,
        },
        Object {
          "opacity": mockConstructor {},
          "transform": Array [
            Object {
              "translateX": mockConstructor {},
            },
            Object {
              "scale": 1,
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F3F3F3",
          "borderRadius": 5,
          "display": "flex",
          "justifyContent": "center",
          "paddingBottom": 13.5,
          "paddingLeft": 37.5,
          "paddingRight": 37.5,
          "paddingTop": 13.5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Raleway_400Regular",
            "fontSize": 40,
          }
        }
      >
        
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 4,
              "display": "flex",
              "height": 20,
              "justifyContent": "center",
              "marginTop": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 4,
              "width": 50,
            },
            Object {
              "backgroundColor": "#4DCC4B",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Raleway_400Regular",
              "fontSize": 20,
            }
          }
        >
          EASY
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
        "marginBottom": 80,
      }
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "opacity": mockConstructor {},
            "transform": Array [
              Object {
                "translateX": mockConstructor {},
              },
              Object {
                "scale": 1,
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "row",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F3F3F3",
                "borderRadius": 5,
                "display": "flex",
                "height": 75,
                "justifyContent": "center",
                "margin": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 75,
              },
              Object {
                "backgroundColor": "#FF9C40",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Raleway_400Regular",
                "fontSize": 40,
              }
            }
          >
            F
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F3F3F3",
                "borderRadius": 5,
                "display": "flex",
                "height": 75,
                "justifyContent": "center",
                "margin": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 75,
              },
              Object {
                "backgroundColor": "#FF9C40",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Raleway_400Regular",
                "fontSize": 40,
              }
            }
          >
            O
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F3F3F3",
                "borderRadius": 5,
                "display": "flex",
                "height": 75,
                "justifyContent": "center",
                "margin": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 75,
              },
              Object {
                "backgroundColor": "#FF9C40",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Raleway_400Regular",
                "fontSize": 40,
              }
            }
          >
            R
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F3F3F3",
                "borderRadius": 5,
                "display": "flex",
                "height": 75,
                "justifyContent": "center",
                "margin": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 75,
              },
              Object {
                "backgroundColor": "#FF9C40",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Raleway_400Regular",
                "fontSize": 40,
              }
            }
          >
            T
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F3F3F3",
                "borderRadius": 5,
                "display": "flex",
                "height": 75,
                "justifyContent": "center",
                "margin": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 75,
              },
              Object {
                "backgroundColor": "#FF9C40",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Raleway_400Regular",
                "fontSize": 40,
              }
            }
          >
            Y
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          undefined,
          Object {
            "opacity": mockConstructor {},
            "transform": Array [
              Object {
                "translateX": mockConstructor {},
              },
              Object {
                "scale": 1,
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          Object {
            "marginTop": -50,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "flexDirection": "row",
              },
              Object {
                "opacity": 0,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#F3F3F3",
                  "borderRadius": 5,
                  "display": "flex",
                  "height": 75,
                  "justifyContent": "center",
                  "margin": 5,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 2,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 4,
                  "width": 75,
                },
                Object {
                  "backgroundColor": "#F3F3F3",
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "Raleway_400Regular",
                  "fontSize": 40,
                }
              }
            >
              F
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#F3F3F3",
                  "borderRadius": 5,
                  "display": "flex",
                  "height": 75,
                  "justifyContent": "center",
                  "margin": 5,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 2,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 4,
                  "width": 75,
                },
                Object {
                  "backgroundColor": "#F3F3F3",
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "Raleway_400Regular",
                  "fontSize": 40,
                }
              }
            >
              O
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#F3F3F3",
                  "borderRadius": 5,
                  "display": "flex",
                  "height": 75,
                  "justifyContent": "center",
                  "margin": 5,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 2,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 4,
                  "width": 75,
                },
                Object {
                  "backgroundColor": "#F3F3F3",
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "Raleway_400Regular",
                  "fontSize": 40,
                }
              }
            >
              R
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#F3F3F3",
                  "borderRadius": 5,
                  "display": "flex",
                  "height": 75,
                  "justifyContent": "center",
                  "margin": 5,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 2,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 4,
                  "width": 75,
                },
                Object {
                  "backgroundColor": "#F3F3F3",
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "Raleway_400Regular",
                  "fontSize": 40,
                }
              }
            >
              T
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#F3F3F3",
                  "borderRadius": 5,
                  "display": "flex",
                  "height": 75,
                  "justifyContent": "center",
                  "margin": 5,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 2,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 4,
                  "width": 75,
                },
                Object {
                  "backgroundColor": "#F3F3F3",
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "Raleway_400Regular",
                  "fontSize": 40,
                }
              }
            >
              Y
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          undefined,
          Object {
            "opacity": mockConstructor {},
            "transform": Array [
              Object {
                "translateX": mockConstructor {},
              },
              Object {
                "scale": 1,
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": mockConstructor {},
                },
              ],
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F3F3F3",
                "borderRadius": 5,
                "display": "flex",
                "height": 75,
                "justifyContent": "center",
                "margin": 5,
                "opacity": 1,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 75,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "Raleway_400Regular",
                  "fontSize": 40,
                }
              }
            >
              F
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": mockConstructor {},
                },
              ],
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F3F3F3",
                "borderRadius": 5,
                "display": "flex",
                "height": 75,
                "justifyContent": "center",
                "margin": 5,
                "opacity": 1,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 75,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "Raleway_400Regular",
                  "fontSize": 40,
                }
              }
            >
              O
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": mockConstructor {},
                },
              ],
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F3F3F3",
                "borderRadius": 5,
                "display": "flex",
                "height": 75,
                "justifyContent": "center",
                "margin": 5,
                "opacity": 1,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 75,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "Raleway_400Regular",
                  "fontSize": 40,
                }
              }
            >
              R
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": mockConstructor {},
                },
              ],
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F3F3F3",
                "borderRadius": 5,
                "display": "flex",
                "height": 75,
                "justifyContent": "center",
                "margin": 5,
                "opacity": 1,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 75,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "Raleway_400Regular",
                  "fontSize": 40,
                }
              }
            >
              T
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": mockConstructor {},
                },
              ],
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F3F3F3",
                "borderRadius": 5,
                "display": "flex",
                "height": 75,
                "justifyContent": "center",
                "margin": 5,
                "opacity": 1,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 75,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "Raleway_400Regular",
                  "fontSize": 40,
                }
              }
            >
              Y
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          undefined,
          Object {
            "opacity": mockConstructor {},
            "transform": Array [
              Object {
                "translateX": mockConstructor {},
              },
              Object {
                "scale": 1,
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "row",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F3F3F3",
                "borderRadius": 5,
                "display": "flex",
                "height": 75,
                "justifyContent": "center",
                "margin": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 75,
              },
              Object {
                "backgroundColor": "#40BAFF",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Raleway_400Regular",
                "fontSize": 40,
              }
            }
          >
            F
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F3F3F3",
                "borderRadius": 5,
                "display": "flex",
                "height": 75,
                "justifyContent": "center",
                "margin": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 75,
              },
              Object {
                "backgroundColor": "#40BAFF",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Raleway_400Regular",
                "fontSize": 40,
              }
            }
          >
            I
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F3F3F3",
                "borderRadius": 5,
                "display": "flex",
                "height": 75,
                "justifyContent": "center",
                "margin": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 75,
              },
              Object {
                "backgroundColor": "#40BAFF",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Raleway_400Regular",
                "fontSize": 40,
              }
            }
          >
            F
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F3F3F3",
                "borderRadius": 5,
                "display": "flex",
                "height": 75,
                "justifyContent": "center",
                "margin": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 75,
              },
              Object {
                "backgroundColor": "#40BAFF",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Raleway_400Regular",
                "fontSize": 40,
              }
            }
          >
            T
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#F3F3F3",
                "borderRadius": 5,
                "display": "flex",
                "height": 75,
                "justifyContent": "center",
                "margin": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 75,
              },
              Object {
                "backgroundColor": "#40BAFF",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Raleway_400Regular",
                "fontSize": 40,
              }
            }
          >
            Y
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "bottom": 200,
          "height": 150,
          "position": "absolute",
        },
        Object {
          "opacity": mockConstructor {},
          "transform": Array [
            Object {
              "translateX": mockConstructor {},
            },
            Object {
              "scale": 1,
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "paddingLeft": 20,
          "paddingRight": 20,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": mockConstructor {},
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F3F3F3",
              "borderRadius": 5,
              "display": "flex",
              "height": 75,
              "justifyContent": "center",
              "margin": 5,
              "opacity": 1,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 4,
              "width": 75,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Raleway_400Regular",
                "fontSize": 40,
              }
            }
          >
            C
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": mockConstructor {},
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F3F3F3",
              "borderRadius": 5,
              "display": "flex",
              "height": 75,
              "justifyContent": "center",
              "margin": 5,
              "opacity": 1,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 4,
              "width": 75,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Raleway_400Regular",
                "fontSize": 40,
              }
            }
          >
            B
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": mockConstructor {},
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F3F3F3",
              "borderRadius": 5,
              "display": "flex",
              "height": 75,
              "justifyContent": "center",
              "margin": 5,
              "opacity": 1,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 4,
              "width": 75,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Raleway_400Regular",
                "fontSize": 40,
              }
            }
          >
            C
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "display": "flex",
        "flexDirection": "row",
        "position": "absolute",
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 26.5,
          "margin": 10,
        }
      }
    >
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": mockConstructor {},
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4294956160,
                  4294950464,
                ],
                "endPoint": undefined,
                "locations": Array [
                  0.5,
                  0.5,
                ],
                "startPoint": Array [
                  -0.1,
                  0.1,
                ],
              }
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 5,
                "display": "flex",
                "height": 106.5,
                "justifyContent": "center",
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 300,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Raleway_700Bold",
                  "fontSize": 33.5,
                }
              }
            >
              RESET
            </Text>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "bottom": 26.5,
          "margin": 10,
        }
      }
    >
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": mockConstructor {},
              },
            ],
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4294956160,
                  4294950464,
                ],
                "endPoint": undefined,
                "locations": Array [
                  0.5,
                  0.5,
                ],
                "startPoint": Array [
                  -0.1,
                  0.1,
                ],
              }
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 5,
                "display": "flex",
                "height": 106.5,
                "justifyContent": "center",
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 4,
                "width": 300,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Raleway_700Bold",
                  "fontSize": 33.5,
                }
              }
            >
              UNDO
            </Text>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
      </View>
    </View>
  </View>
</View>
`;
